File: MusicManagerAPI/src/main/java/com/ldz/music/manager/service/IMusicManagerService.java
package com.ldz.music.manager.service;

import com.ldz.music.manager.constants.MusicTypes;
import com.ldz.music.manager.model.bo.UserMusicStatusBO;

/**
 * Created by ldalzotto on 26/04/2017.
 */
public interface IMusicManagerService {

    public void addUserMusicStatus(UserMusicStatusBO userMusicStatusBO);

    public void deleteUserMusicStatusFromUsername(String username);

    public UserMusicStatusBO getMusicStatusFromUsername(String username);

    public String getSourceurlFromUrlAndMusicType(String url, MusicTypes musicType);

    public boolean postMusicFromUsernameAndSourceurl(String username, String sourceUrl);

}

File: MusicManagerAPI/src/scala/com/ldz/music/manager/service/IMusicManagerService.scala
package com.ldz.music.manager.service

import com.ldz.enumeration.ExternalMusicKey
import com.ldz.music.manager.constants.MusicTypes
import com.ldz.music.manager.model.bo.{MusicTypeBO, UserMusicStatusBO}

/**
  * Created by Loic on 20/06/2017.
  */
trait IMusicManagerService {
  def addUserMusicStatus(userMusicStatusBO: UserMusicStatusBO): Unit

  def deleteUserMusicStatusFromUsername(username: String): Unit

  def getMusicStatusFromUsername(username: String): UserMusicStatusBO

  def getMusicparametersFromUrlAndMusicType(url: String, musicType: MusicTypes.Value): MusicTypeBO

  def postMusicFromUsernameAndSourceurl(username: String, musicDetail: MusicTypeBO): MusicTypeBO
}
